addons/survey/models/survey_question.py

```diff
@@ -264,6 +265,36 @@ def _compute_save_as_email(self):
             if question.question_type != 'char_box' or not question.validation_email:
                 question.save_as_email = False
 
+    @api.depends('survey_id.session_speed_rating_time_limit', 'survey_id.session_speed_rating', 'is_time_limited')
+    def _compute_time_limit(self):
+        """If time limit value is set manually on the question, we do not update it
+        when changing the time limit value of the survey or disabling it.
+        """
+        for question in self:
+            if not question.is_time_limited:
+                question.time_limit = False
+            if not question.survey_id.session_speed_rating:
+                if question.time_limit == question.survey_id.session_speed_rating_time_limit:
+                    question.time_limit = False
+                    question.is_time_limited = False
+            elif question.is_time_limited:
+                if question.time_limit == 0 or question.time_limit == question.old_time_limit:
```

Comment:
Isn't it possible to use _origin and have the original survey value ? Don't really like adding a field just for an onchange-like behavior ...

Reply:
Also: once you tick "is_time_limited" on a question, not sure we want to distinguish False and 0, makes code hard to understand and predict...

Reply:
wasn't able to make _origin work, I'll look into this part more closely and see if I can come up with  better solution...

Reply:
_origin works for most cases but it goes awry if someone changes the default limit straight away, and sometimes does not update while changing the time_limit value, which makes the whole thing pretty buggy. So For I think we can either add a new field like before, or do not change the question.time_limit at all and make the answer score calculation dependent on both survey time limit and question time limit separately.